hangman/lib/game/game.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Hangman.Game do
+  @moduledoc """
+  Implemnation of Hangman API.
+  """
+
+  #file with helper methods
+  import Hangman.Assist
+
+  #state structure
+  defstruct turns_left: 7,
+            game_state: :initializing,
+            letters: [],
+            used: [],
+            word: "",
+            last_guess: ""
+  
+  #new_game() method that takes a random word and uses it  
+  def new_game() do
+    word = Dictionary.random_word()
+    %Hangman.Game{}
+    |> Map.put(:word, word)
+    |> Map.put(:letters, List.duplicate("-", String.length(word)))
+  end
+
+  #tally method that returns state without word
+  def tally(game), do: Map.from_struct(game) |> Map.delete(:word)
```

Comment:
## -2

I really wanted the tally to be a separate structure, initializd from the game state. That way it won't accidentally leak values if you add fields to the state later.